src/dra/6D59C.c

```diff
@@ -71,7 +71,68 @@ void func_8010D59C(void) {
     }
 }
 
-INCLUDE_ASM("dra/nonmatchings/6D59C", func_8010D800);
+u8 g_D_800ACF2C[] = {4, 4, 4, 4, 6, 6, 6, 6, 8, 8, 16, 16, 16, 16, 16, 16};
+u8 g_D_800ACF3C[] = {8,  12, 16, 20, 24, 28, 32, 32,
+                     32, 32, 32, 32, 32, 32, 32, 32};
+
+void func_8010D800(void) {
+    byte pad[0x28];
+    Unkstruct_800ECBF8_1* current_thing;
+    Primitive* prim;
+    s32 entNum;
+    s32 i;
+    u8 temp_t0;
+    u8 temp_t1;
+    u8 temp_t2;
+
+    i = 0;
+    prim = &g_PrimBuf[g_Entities[1].primIndex];
```

Comment:
`g_Entities[ENTITY_ALUCARD_SHADOW].primIndex` or similar?

Reply:
Could you try checking if this also applies to Richter when you do the dash move? I'd do it myself but I don't know how to nop out functions.

Reply:
You can do it by going  to the beginning of the function and replacing the first 8 bytes with `0800E003 00000000`
```
/* 0800E003 */  jr         $ra
/* 00000000 */   nop
```

Reply:
This function doesn't seem to execute at all during the prologue so I think Richter's shadows are handled somewhere else.

Reply:
Perfect, thanks! Richter's shadows appear to be unaffected when changing this. In the prologue I saw some activity in the region of g_Entities[1], but this happened regardless of whether he had shadows happening, so I'm not sure how that works.

I'm a little concerned at how SD's function seems to think it's taking control from the player, but this one seems to be about shadows. I think overall we still need to learn more about g_Entities[1] before renaming it to ENITTY_ALUCARD_SHADOW or anything else.